src/hiero_sdk_python/staking_info.py

```
@@ -0,0 +1,132 @@
+        if self.staked_node_id is not None:
+            proto.staked_node_id = self.staked_node_id
+
+        return proto
+
+    @classmethod
+    def from_bytes(cls, data: bytes) -> "StakingInfo":
+        """
+        Creates a StakingInfo instance from protobuf-encoded bytes.
+        """
+        if not isinstance(data, bytes):
+            raise ValueError("data must be bytes")
```

Ruff (0.14.13)

Prefer `TypeError` exception for invalid type

(TRY004)